packages/components/pie-notification/src/index.ts

```diff
@@ -144,87 +118,53 @@ export class PieNotification extends LitElement implements NotificationProps {
      *
-     * @param {NonNullable<NotificationProps['variant']>} variant
-     *
      * @private
      */
-    private getDefaultVariantIcon (variant: NonNullable<NotificationProps['variant']>) {
-        switch (variant) {
+    private getDefaultVariantIcon () {
+        switch (this.variant) {
             case 'info':
-                return html`<icon-info-circle size="m" data-test-id="${componentSelector}-heading-icon-info"></icon-info-circle>`;
+                return html`<icon-info-circle class="icon" size="m" data-test-id="${componentSelector}-heading-icon-info"></icon-info-circle>`;
             case 'success':
-                return html`<icon-check-circle size="m" data-test-id="${componentSelector}-heading-icon-success"></icon-check-circle>`;
+                return html`<icon-check-circle class="icon" size="m" data-test-id="${componentSelector}-heading-icon-success"></icon-check-circle>`;
             case 'warning':
-                return html`<icon-alert-triangle size="m" data-test-id="${componentSelector}-heading-icon-warning"></icon-alert-triangle>`;
+                return html`<icon-alert-triangle class="icon" size="m" data-test-id="${componentSelector}-heading-icon-warning"></icon-alert-triangle>`;
             case 'error':
-                return html`<icon-alert-circle size="m" data-test-id="${componentSelector}-heading-icon-error"></icon-alert-circle>`;
+                return html`<icon-alert-circle class="icon" size="m" data-test-id="${componentSelector}-heading-icon-error"></icon-alert-circle>`;
             default:
                 return nothing as never;
         }
     }
 
-    /**
-     * Util method that returns a template with a default icon according to the chosen variant.
-     * Called within the renderIcon method.
-     *
-     * @param {NonNullable<NotificationProps['variant']>} variant
-     *
-     * @private
-     */
-    private renderIconVariant (variant: NonNullable<NotificationProps['variant']>) {
-        if (this.variantHasDefaultIcon(variant)) {
-            return this.getDefaultVariantIcon(variant);
-        }
-
-        return nothing;
-    }
-
     /**
      * Template for the heading icon area.
-     * It can return an icon provided externally via named slot or it can return an default icon according to the chosen variant.
+     * It can return an icon provided externally via named slot or it can return a default icon according to the chosen variant if defined.
      * Called within the main render function.
      *
-     * @param {NonNullable<NotificationProps['variant']>} variant
-     * @param {boolean} hasExternalIcon
-     *
      * @private
      */
-    private renderIcon (variant: NonNullable<NotificationProps['variant']>, hasExternalIcon: boolean, hasIconClass: boolean): TemplateResult | typeof nothing {
+    private renderIcon (): TemplateResult | typeof nothing {
+        const hasExternalIcon = this._iconSlot.length > 0;
```

Comment:
the value of `this._iconSlot.length` is reactive so we can listen to it directly to render the default icon for an icon if defined.

Reply:
If this is something a bit out of the ordinary, is it worth a comment in the code?

Reply:
I don't think it's out of the ordinary, it's probably just we didn't know it's the case. 

the lit docs already go over how to query slot elements [here](https://lit.dev/docs/api/decorators/#queryAssignedElements)